test_programs/compile_success_empty/serialize/src/main.nr

```diff
@@ -0,0 +1,59 @@
+trait Serialize {
+    let Size: u32;
+
+    // Note that Rust disallows referencing constants here!
+    fn serialize(self) -> [Field; Self::Size];
+}
+
+impl<A, B, let AS: u32, let BS: u32> Serialize for (A, B) where A: Serialize<Size = AS>, B: Serialize<Size = BS> {
+    // let Size = <A as Serialize>::Size + <B as Serialize>::Size;
```

Comment:
Doesn't work?

Reply:
All associated types must be explicit currently so this would have to be rewritten as
```rs
let Size = <A as Serialize<Size = AS>>::Size + <B as Serialize<Size = BS>>::Size;
```
which is somewhat pointless when you're already specifying AS and BS. This limitation is mentioned in the docs as well.

There's another issue in that the `<T as Trait>::Type` syntax isn't allowed in arithmetic generics currently. I plan to make an issue for this.